src/Yaapii.Atoms/Number/NumberOf.cs

```diff
@@ -38,149 +38,173 @@ public sealed class NumberOf : NumberEnvelope
-                    text,
-                    new NumberFormatInfo()
-                    {
-                        NumberDecimalSeparator = decimalSeperator,
-                        NumberGroupSeparator = blockSeperator
-                    }
-                )
-            ),
-            new ScalarOf<long>(() =>
-                Convert.ToInt64(
-                    text,
-                    new NumberFormatInfo()
-                    {
-                        NumberDecimalSeparator = decimalSeperator,
-                        NumberGroupSeparator = blockSeperator
-                    }
-                )
-            ),
-            new ScalarOf<float>(() =>
-                (float)Convert.ToDecimal(
-                    text,
-                    new NumberFormatInfo()
-                    {
-                        NumberDecimalSeparator = decimalSeperator,
-                        NumberGroupSeparator = blockSeperator
-                    }
-                )
-            )
+        public NumberOf(string text, string decimalSeperator, string blockSeperator) : this(
+            new ScalarOf<long>(() => Convert.ToInt64(
+                text,
+                new NumberFormatInfo()
+                {
+                    NumberDecimalSeparator = decimalSeperator,
+                    NumberGroupSeparator = blockSeperator
+                })),
+            new ScalarOf<int>(() => Convert.ToInt32(
+                text,
+                new NumberFormatInfo()
+                {
+                    NumberDecimalSeparator = decimalSeperator,
+                    NumberGroupSeparator = blockSeperator
+                })),
+            new ScalarOf<float>(() => (float)Convert.ToDecimal(text, new NumberFormatInfo()
+            {
+                NumberDecimalSeparator = decimalSeperator,
+                NumberGroupSeparator = blockSeperator
+            })),
+            new ScalarOf<double>(() => Convert.ToDouble(
+                text,
+                new NumberFormatInfo()
+                {
+                    NumberDecimalSeparator = decimalSeperator,
+                    NumberGroupSeparator = blockSeperator
+                }))
         )
         { }
 
         /// <summary>
         /// A <see cref="int"/> as a <see cref="INumber"/>
         /// </summary>
         /// <param name="str">The string</param>
-        public NumberOf(string str) : this(str, new ScalarOf<IFormatProvider>(() => CultureInfo.InvariantCulture))
+        public NumberOf(string str) : this(str, CultureInfo.InvariantCulture)
         { }
 
+
         /// <summary>
-        /// A <see cref="int"/> as a <see cref="INumber"/>
+        /// A <see cref="string"/> as a <see cref="INumber"/>
         /// </summary>
         /// <param name="str">The string</param>
         /// <param name="provider">a number format provider</param>
-        public NumberOf(string str, IFormatProvider provider) : this(str, new ScalarOf<IFormatProvider>(provider))
```

Comment:
If you remove this ctor, it makes this a breaking change. So please preserve the ctor and add another one without a scalar if it is needed.